### ISSUE TITLE
@me is  not retrieving the current user's email. 

### ISSUE THREAD

**Issue by CCC-Jonathan** [SharePoint/sp-dev-docs#6516]
We are looking to use the @me as documented on this page.  But it doesn't work.  even if we try to spit it out as content we just get nothing.
txtContent": "=@me"
just gives use the literal text "@me" on the screen.  It doesn't seem to be interrupting @me as a command.  
Any suggestions?

Thank-you
Jonathan

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 002a1ee9-dc9f-8dc1-6f3c-ef23297b83d4
* Version Independent ID: 82565c7e-b264-94f5-8708-d4757482e133
* Content: [Use view formatting to customize SharePoint](https://docs.microsoft.com/en-us/sharepoint/dev/declarative-customization/view-formatting)
* Content Source: [docs/declarative-customization/view-formatting.md](https://github.com/SharePoint/sp-dev-docs/blob/master/docs/declarative-customization/view-formatting.md)
* Product: **sharepoint**
* GitHub Login: @spdevdocs
* Microsoft Alias: **spdevdocs**

**Comment by Jerry0527** [comment]
Try to modify from "txtContent": "=@me" to  "txtContent": "@me", this is the test formatting:

```
{
   "$schema": "https://developer.microsoft.com/json-schemas/sp/v2/column-formatting.schema.json",
   "elmType": "div",
   "txtContent": "@me"
}
```

![Uploading Snipaste_2020-12-02_08-52-33.png…]()

**Comment by CCC-Jonathan** [comment]
![ScreenShotAtMe](https://user-images.githubusercontent.com/75339098/100818614-bcb00600-3407-11eb-82b3-72922304e372.png)
Hi Jerry,
Thank-you for your quick response.
I have tried that, and it still just spits it out as a string instead of interrupting it.

**Comment by Jerry0527** [comment]
What's your SharePoint version ? I'm testing in SharePoint Online and it's working as expected.
![Uploading Snipaste_2020-12-02_10-10-49.png…]()

**Comment by CCC-Jonathan** [comment]
I also have:
"txtContent": "=if ([$BidderNo] == @me, 'You are the Highest bidder', [$BidderNo])

Where $BidderNo is a column in my list that contains the users email addresses.  my If is never evaluating to true.

I believe I am using Sharepoint online: https://_________________________.sharepoint.com/sites/_________________________/Lists/AuctionApp/Live%20Auction%20Display.aspx
![ScreenShotSharepoint](https://user-images.githubusercontent.com/75339098/100819585-af941680-3409-11eb-9ea2-a47a68bcf51b.png)

**Comment by Jerry0527** [comment]
I tested the if conidtion formatting in another thread posted by you [here](https://github.com/SharePoint/sp-dev-docs/issues/6517), still works please see the capture below:

```
{
"elmType": "div",
"attributes": {
"class": "sp-card-displayColumnContainer"
},
"children": [
{
"elmType": "p",
"attributes": {
"class": "ms-fontColor-neutralPrimary sp-card-content sp-card-highlightedContent"
},
"txtContent": "=if ([$BidderNo] == @me, 'You are the Highest bidder', [$BidderNo])"
}
]
}
```
![Snipaste_2020-12-02_10-20-42](https://user-images.githubusercontent.com/28797671/100819894-0b04e980-3488-11eb-9f44-f8df8e5c079d.png)

Can you create a new List and try to test the formatting ? As it's actually working in my side.

**Comment by CCC-Jonathan** [comment]
Hi Jerry,

That is a good idea.  I will create a brand new list and test it.  Thank-you for the suggestion.  I will let you know.

**Comment by CCC-Jonathan** [comment]
I created a new list in my site.  I choose a blank list instead of using any default template.  I then Added three items.  From the View menu, I choose "View Tiles".  Then clicked Format Current View.  Under Choose Layout, I selected "Gallery".  This dumped in the preset JSON.  I simply added:
		  {
            "$schema": "https://developer.microsoft.com/json-schemas/sp/v2/column-formatting.schema.json",
            "elmType": "div",
            "txtContent": "@me"
          }

	To the preset JSON (had to increase the height of the initial div)
	But it just spit out the string @me, it did not interrupt retrieve the user's email.
![ScreenShotSharepointNewList](https://user-images.githubusercontent.com/75339098/100898045-94fb8500-347d-11eb-954a-69617911f2aa.png)

**Comment by Jerry0527** [comment]
I tested the sample code from official document [here](https://docs.microsoft.com/en-us/sharepoint/dev/declarative-customization/view-formatting) to format a view:

```
{
  "$schema": "https://developer.microsoft.com/json-schemas/sp/view-formatting.schema.json",
  "hideSelection": true,
  "hideColumnHeader": true,
  "rowFormatter": {
    "elmType": "div",
    "attributes": {
      "class": "sp-row-card"
    },
    "children": [
      {
        "elmType": "div",
        "style": {
          "text-align": "left"
        },
        "children": [
          {
            "elmType": "div",
            "attributes": {
              "class": "sp-row-title"
            },
            "txtContent": "@me"
          }
        ]
      }
    ]
  }
}
```
It's still working as expected:

![Snipaste_2020-12-03_08-35-34](https://user-images.githubusercontent.com/28797671/100948372-ad81a300-3542-11eb-8436-f640a5678a86.png)

I suggest you can have a try.

If the issue still existed, try @me in Format columns:


```
{
   "$schema": "https://developer.microsoft.com/json-schemas/sp/v2/column-formatting.schema.json",
   "elmType": "div",
   "txtContent": "@me"
}
```

**Comment by CCC-Jonathan** [comment]
Hi Jerry,
Thank-you for your suggestions.  Unfortunately, I still get the plain text and it is not being interrupted.
![ScreenShotSharepointNewList2](https://user-images.githubusercontent.com/75339098/100950457-9ce11680-34c9-11eb-9d5a-2521ee3d4442.png)


Is there maybe a Site wide setting that needs to be set to tell the JSON to interrupt @me?  because the JSON just keeps reading it as a literal string instead of interrupting the @ as a command.

**Comment by CCC-Jonathan** [comment]
When I change @me to @CurrentField.title
"txtContent": "@currentField.Title"
It is blank.  

Interestingly, when I use "@currentField", then I do get the Title to show.  So that is good. And it means that the @ is seen as a special character, and being interrupted as a command.  But @me does not work.
@now works.
@currentWeb works as well
@rowIndex works
@window.innerWidth works
But @me does not work.

Any suggestions that you may have would be greatly appreciated.
Thank-you
Jonathan

**Comment by Jerry0527** [comment]
"@" should be working correctly in Json Column Formatting rather than a special character.
And there is no settings for the "@me" in Site Wide.
Did the current user have valid email address ? Actually I can't reproduce your issue  in my side.

**Comment by Jerry0527** [comment]
I mean the valid Email address should have the Exchange Online (Plan2) license:

![Snipaste_2020-12-03_16-59-50](https://user-images.githubusercontent.com/28797671/100987063-f8260e00-3588-11eb-8c75-cc1244fdd194.png)

Please check this point in Microsoft 365 Admin Center-> Users->Active Users:


![Snipaste_2020-12-03_17-01-24](https://user-images.githubusercontent.com/28797671/100987209-2f94ba80-3589-11eb-9064-ffafcdaf62d5.png)

If not assiging this license in your side, then please assign this to the user. 

Waiting  for  a day to make changes sync, then this should be able to use @me to get the email address.

**Comment by CCC-Jonathan** [comment]
I have been able to get 
  User().Email
to work in PowerApps Forms
And I am able to use User Email in MS Flow.

I have tried from 3 different user accounts (from 3 different organizations).  and for none of them the @me works.

Under my licenses, I only had the option for Exchange Forms (Plan E1) and that was unchecked.  I have checked that.  And will wait for a day.  Hopefully this will help.

Thank-you very much for your insights and help.
